task6.rb

```diff
@@ -1,8 +1,18 @@
+class NotANumberError < StandardError
+  def initialize(obj)
+    super("#{obj} not a number")
+  end
+end
+
 class PseudoInt
 
   attr_reader :num
   def initialize(num)
-    @num = num
+    if num.is_a?(Numeric)
```

Comment:
this validation is not actually working as if you
`'qqq'.to_i` will produce 0
You can use [RegExp](https://www.rubyguides.com/2015/06/ruby-regex/) to validate your input params